everestrs/everestrs_sys/everestrs_sys.cpp

```diff
@@ -0,0 +1,83 @@
+#include "everestrs_sys.hpp"
+
+#include <cstdlib>
+#include <stdexcept>
+
+#include "everestrs/lib.rs.h"
+
+namespace {
+
+std::unique_ptr<Everest::Everest> create_everest_instance(
+    const std::string& module_id, const Everest::RuntimeSettings& rs,
+    const Everest::Config& config) {
+  return std::make_unique<Everest::Everest>(
+      module_id, config, true /* FIXME */, rs.mqtt_broker_host,
+      rs.mqtt_broker_port, rs.mqtt_everest_prefix, rs.mqtt_external_prefix,
+      rs.telemetry_prefix, rs.telemetry_enabled);
+}
+
+std::unique_ptr<Everest::Config> create_config_instance(
+    const Everest::RuntimeSettings& rs) {
+  // FIXME (aw): where to initialize the logger?
+  Everest::Logging::init(rs.logging_config_file);
+  return std::make_unique<Everest::Config>(
+      rs.schemas_dir.string(), rs.config_file.string(), rs.modules_dir.string(),
+      rs.interfaces_dir.string(), rs.types_dir.string(), rs.mqtt_everest_prefix,
+      rs.mqtt_external_prefix);
+}
+
+JsonBlob json2blob(const json& j) {
+  // I did not find a way to not copy the data at least once here.
+  const std::string dumped = j.dump();
+  rust::Vec<uint8_t> vec;
+  vec.reserve(dumped.size());
+  std::copy(dumped.begin(), dumped.end(), std::back_inserter(vec));
```

Comment:
not really what you want but std::move should make one copy less

Reply:
Not sure where the std::move would need to go here... std::back_inserter is taking a reference already and the move into the `JsonBlob` is elided by the compiler anyways, see https://godbolt.org/z/986cWfMqs.